### ISSUE TITLE
Android data base inconsistent - comparing  two (Samsung) devices

### ISSUE THREAD

**Issue by gubrist** [Automattic/simplenote-android#669]
#### Steps to reproduce
1. Use Simplenote Version 1.7.0 on Android 7.1.1 
on Samsung Galaxy Tab A (9.7, LTE) SM-T555 
which shows ALL Notes

2. Use Simplenote Version 1.7.0 on Android 8.1.0 
on Samsung Galaxy XCover 4 (SM-G390F), 
which shows NOT ALL Notes

#### What I expected
Simplenote that makes shure and 
confirms e.g. with a synch log that ALL Notes are available. 

#### What happened instead
2. NOT ALL Notes are available

#### Simplenote version
Simplenote Version 1.7.0 

#### OS version
see above

#### Screenshot / Video

App Memory of 1:

![Screenshot_20190430-120213](https://user-images.githubusercontent.com/22197180/56989930-f49ff900-6b93-11e9-910b-15e4ebec5bf2.jpg)

App Memory of 2:

![Screenshot_20190430-115928_Settings](https://user-images.githubusercontent.com/22197180/56992643-ce318c00-6b9a-11e9-91b8-f83460b18438.jpg)

**Comment by theck13** [comment]
@gubrist, are all the notes on both devices?  I wouldn't be surprised if the storage amounts are different between a phone and tablet running different Android versions, but all the notes are still there.

**Comment by gubrist** [comment]
@theck13 
No: Samsung Galaxy XCover 4 (SM-G390F), does NOT show ALL Notes.

**Comment by gubrist** [comment]
@theck13 
No: Samsung Galaxy XCover 4 (SM-G390F), does NOT show ALL Notes.

**Comment by theck13** [comment]
I'm not sure how the devices became out-of-sync, but you can try to resolve the difference to see if it was a fluke or consistent.  If the Samsung Galaxy Tab A (SM-T555) has _*all*_ the notes, then I would log out of Simplenote on the Samsung Galaxy XCover 4 (SM-G390F), clear the app cache, clear the app data, then log in again.  _*WARNING*_:  Only do that if there are no notes on the Samsung Galaxy XCover 4 (SM-G390F) that have not been synced.  Otherwise, you'll lose the notes when logging out and clearing the app cache/data.

**Comment by gubrist** [comment]
I have opened this:

Number of notes per tag (notes count) #687

https://github.com/Automattic/simplenote-android/issues/687

**Comment by theck13** [comment]
The number of tags per now is already available so I closed https://github.com/Automattic/simplenote-android/issues/687.

**Comment by gubrist** [comment]
> I would log out of Simplenote on the Samsung Galaxy XCover 4 (SM-G390F), clear the app cache, clear the app data, then log in again. 

Thanks, done. 

I have moved the App to the SD Card. 

How can I properly check if the data base is consistent? 

This Option does not work:

https://github.com/Automattic/simplenote-android/issues/687

**Comment by theck13** [comment]
The web app will show the correct data.  Check app.simplenote.com to see what notes and tags are stored in the database.

**Comment by gubrist** [comment]
1. I had my  Samsung Galaxy XCover 4 (SM-G390F) connected with WiFi  during 30 minutes to download all noted.

2. I have **changed** an existing note.

3. Waited  30 minutes ---> Note on web app not changed

4. I have created a **new** note.

5. Waited  15 minutes ---> Note on web app not created.

**Comment by gubrist** [comment]
The **turning wheel** appeared right after 2.

![wheel](https://user-images.githubusercontent.com/22197180/58729251-36b59800-83e9-11e9-9a9f-84ed77d3fe90.JPG)

What is the meaning of this **turning wheel**?
Is this a clue to the problem?

**Comment by theck13** [comment]
If your notes are not showing on the web, then there is a problem with your device(s).  Enter the notes on the web to ensure they are being saved to the database.  Once all your notes are in the web, log out of your device(s) and clear the cache/data using the Android Settings app.  Then log back in on your device(s).

If you're still having problems, contract us at support@simplenoteapp.com and reference this issue.

**Comment by gubrist** [comment]
> If you're still having problems, contact us at [support@simplenoteapp.com] and reference this issue.

Thanks. I have done that. Now, approx. ten emails later, having explained (again) the issue to various persons, I have received more or less the same answers as stated here.

**Threre is currently no way to find out if my > 4500 notes are really downloaded from the Web App (and consequently uploaded) .**

**Comment by bummytime** [comment]
@gubrist — we are working directly with our support staff on this issue so I am closing this issue. Going forward, please direct all communication there. Thank you!

**Comment by gubrist** [comment]
**This sounds promising**

Rosie M. (Automattic)

Jan 8, 2021, 16:12 UTC

The Android app currently only syncs when it's in the foreground, meaning when you have the app showing on the screen. When you lock your screen or open another app the sync stops.

We will be changing this in the Simplenote Android 2.15.0 release so it syncs in the background as well.

Also, currently, there is no definitive way to say sync is complete on Android. We just merged a feature to show the last synced time in the Information sheet on Android and that will be available in version 2.15 as well.

**Comment by gubrist** [comment]
I did the following with my three Samsung Andorid devices, with two days full attention (close next to WiFi access):

- Log off my Simplenote Account
- Delete App Cache
- Delete App Data
- Update to Android 2.15.

The two "older" devices:

a) Samsung Galaxy XCover 4 (SM-G390F) and
Android Version: 9

n) Samsung Galaxy Tab A (9.7, LTE) SM-T555
Android Version: 7

- do not upload new notes created on the device
- do not download all existing notest (data base still inconsistent, not reliable)

My newest 
Samsung Galaxy S9 Plus SM-G965F (64 GB)  
Android Version: 10
seems to cope well.

I assume it is an issue with the energy saving options, which depend not just on the Android version but directly to the hardware model.

**Comment by gubrist** [comment]
I have purchased yesterday a brandnew

Samsung Galaxy A51 - SM-A515F
with 128 GB Memory

Energy saving mode is off.

Android 10, Simplenote 2.15 (130)

- A new, edited note is not uploaded
- Lots of exisiting notes are not downloaded

Update 27.02.2021
---> The complete **download** took some hours / days.
---> **Upload** after device was not online when creating/chaning needs to be investigated (by me).

I whished the App would realize and confirm when not all notes have been synched.
